torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -451,6 +451,350 @@ class IrParser {
+            /*
+            auto current_var_hat = mul(var, new Float(kMomentum));
+            auto rvar_bcast = broadcast(running_var, broadcast_mask);
+            auto var_hat = mul(rvar_bcast, new Float(1.0 - kMomentum));
+            auto new_var_hat = add(var_hat, current_var_hat);
+            */
+
+            auto var_eps = add(var, new Float(kEps));
+            auto rvar = unaryOp(UnaryOpType::Rsqrt, var_eps);
+            auto output = mul(x_mean_sub, rvar);
+
+            std::vector<TensorView*> all_tensors({input,
+                                                  // current_mean,
+                                                  // rmean_bcast,
+                                                  // mean_hat,
+                                                  // current_mean_hat,
+                                                  // current_var,
+                                                  // rvar_bcast,
+                                                  // var_hat,
+                                                  // current_var_hat,
+                                                  x_sum,
+                                                  var_sum,
+                                                  x_mean,
+                                                  x_sum_bcast,
+                                                  x_mean_sub,
+                                                  x_mean_sub_pow,
+                                                  var_sum_bcast,
+                                                  var,
+                                                  var_eps,
+                                                  rvar,
+                                                  output});
+
+            // Optional: norm * weight
+            if (weight) {
+              auto weight_bcast = broadcast(weight, broadcast_mask);
+              output = mul(output, weight_bcast);
+              all_tensors.push_back(weight_bcast);
+              all_tensors.push_back(output);
+            }
+
+            // Optional: norm * weight + bias
+            if (bias) {
+              auto bias_bcast = broadcast(bias, broadcast_mask);
+              output = add(output, bias);
+              all_tensors.push_back(bias_bcast);
+              all_tensors.push_back(output);
+            }
+            value_map.emplace(node->output()->unique(), output);
+
+            // Merge reduction axes together
+            // Reorder axes such that reduction axis is at inner-most
+            // dimension [Outer, Reduction]
+            for (auto tv : all_tensors) {
+              // Transform [outer, C, inner] to [C, outer, inner]
+              tv->reorder({{0, 1}, {1, 0}});
+
+              // Transform [C, outer, inner] to [C, reduction]
+              // Flatten reduction axis together
+              for (int idx = 1; idx < kNumberOfDims - 1; ++idx) {
+                tv->merge(1, 2);
+              }
+            }
+          });
+    }
```

Comment:
We should mark normalization as (reduction) operations: https://github.com/csarofeen/pytorch/blob/52581a0bf13f1a92e8346e6fbb08c7f44fcb9d55/torch/csrc/jit/codegen/cuda/parser.cpp#L153-L157